File: Practice/07/Phyton/777.py
import math
print ("Выберите способ")
x = int(input())
p = float
s = float
if x == 1:
    print('Введите длины сторон')
    a = int(input())
    b = int(input())
    c = int(input())
    p=(a*b*c)/2
    s=math.sqrt(p*(p-a)*(p-b)*(p-c))
    print (s)
elif x == 2:
    print('Введите координаты вершин')
    x1, y1, x2, y2, x3, y3 = list(map(int, input().split()))
    a = math.sqrt((x2 - x1) * 2 + (y2 - y1) * 2)
    b = math.sqrt((x3 - x1) * 2 + (y3 - y1) * 2)
    c = math.sqrt((x2 - x3) * 2 + (y2 - y3) * 2)
    p = (a + b + c) / 2
    if p <= 0:
         print ("Ошибочный ввод")
    else:
        s = math.sqrt(p * (p - a) * (p - b) * (p - c))
        print (s)
else:
    print('Ошибочный ввод')
    

File: Practice/07/C++/777.cpp

#include <iostream>
#include <math.h>

using namespace std;
double a, b, c, x, p, s;
int q, w, e, r, d, f;
int main()
{
    std::cout << ("Выберете способ") << "\n";
    std::cin >> x;
    if (x == 1)
    {
        std::cout << ("Введите длинны сторон") << "\n";
        std::cin >> a;
        std::cin >> b;
        std::cin >> c;
        p = (a + b + c) / 2;
        s = sqrt(p * (p - a) * (p - b) * (p - c));
        std::cout << (s) << "\n";
    }
    else
    {
        std::cout << ("Введите координаты вершин") << "\n";
        std::cout << "\n x1="; std::cin >> q;
        std::cout << "\n x2="; std::cin >> w;
        std::cout << "\n x3="; std::cin >> e;
        std::cout << "\n y1="; std::cin >> r;
        std::cout << "\n y2="; std::cin >> d;
        std::cout << "\n y3="; std::cin >> f;
        a = sqrt((q - w) ^ 2 + (r - d) ^ 2);
        b = sqrt((q - e) ^ 2 + (r - f) ^ 2);
        c = sqrt((w - e) ^ 2 + (d - f) ^ 2);
        p = (a + b + c) / 2;
        s = sqrt(p * (p - a) * (p - b) * (p - c));
        std::cout << (s) << "\n";

    }
    if (s <= 0)
        std::cout << "\n Неверный ввод";

}
